### PR TITLE
Embedded Auth PoC

### PR DESCRIPTION
**Pull request by jameskraus**
Proof of concept of an embedded authentication flow. Opens an embedded webview and uses that to scrape a valid JWT from the user's session. From there, it's possible to get their user id and subsequently query their username, known words, etc. 

Preview:
smaller

### CODE REVIEW

File: duolingo_sync/plugin.py [hunk 1]
```diff
@@ -64,14 +64,12 @@ def login_and_retrieve_vocab(username, password) -> VocabRetrieveResult:
                 """
                 <p>Logging in to Duolingo failed. Please check your Duolingo credentials.</p>
 
-                <p>Having trouble logging in? You must use your <i>Duolingo</i> username and password.
```
Comment (jameskraus) [inline duolingo_sync/plugin.py:67]: Edited this copy in https://github.com/JASchilz/AnkiSyncDuolingo/pull/65/files, which I based this PR on. Not sure this copy really makes sense in the context of embedded authentication.

File: duolingo_sync/plugin.py [hunk 2]
```diff
@@ -64,14 +64,12 @@ def login_and_retrieve_vocab(username, password) -> VocabRetrieveResult:
...
+                <p>Having trouble logging in? You must use your <i>Duolingo</i> username and JWT.
                 You <i>can't</i> use your Google or Facebook credentials, even if that's what you use to
```
Comment (jameskraus) [inline duolingo_sync/plugin.py:68]: Interestingly I believe Google / Facebook creds will now work with embedded auth.

File: duolingo_sync/duolingo.py [hunk 3]
```diff
@@ -1,577 +1,576 @@
...
-    setattr(Duolingo, attr, prop)
-
-if __name__ == '__main__':
-    from pprint import pprint
-
-    duolingo = Duolingo('ferguslongley')
-    knowntopic = duolingo.get_known_topics('it')
-
-    pprint(knowntopic)
+"""
```
Comment (jameskraus) [inline duolingo_sync/duolingo.py:1]: It seems the line endings on this file changed, resulting a very large diff

**Comment by JASchilz** [comment]
Just wanted to let you know that I am following and appreciate the request,
and will be able to take a deeper looking this weekend. :)

**Comment by JASchilz** [comment]
Hey James. This works really beautifully, and is an improvement on the old in-Anki username/password scheme that it's replacing and is much appreciated.

I'm going to:
  * Merge this into a branch in this repository `embedded-auth`.
  * Put out a release candidate for anyone who is following along and would like to install from a file.
  * Then add a few more things before I make a full release: like a pop up with the links to get help.

I suspect I'll be able to get a full release out today, but it might take me a couple of extra days.

On a related note:
  * Duolingo has made a couple of breaking changes to their APIs in the last few months.
  * I don't actually use this plugin myself anymore.
  * And so I'm feeling like it might not be feasible for me to continue play cat-and-mouse with them in the future if they continue to make more API changes.

So I think that if there's another breaking change that comes down the pipe, I'll probably take the plugin off the Anki site and just release directly off of GitHub, advising users to expect less support (or community support!) due to that API volatility.

**Comment by jameskraus** [comment]
Sounds like a plan. Thanks for pushing this forward.

> * Duolingo has made a couple of breaking changes to their APIs in the last few months.
> * I don't actually use this plugin myself anymore.

Yeah, I saw the breaking changes with the translations, but it's not too bad given that we can supplement with Google Translate / DeepL translation plugins. I totally get that the maintenance burden can be high if you're not actively using it this plugin. It's funny, since I think Duolingo kind of naturally lends itself towards using it for a bit and then dropping off.

**Pull request merged**